sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningAwareFileIndex.scala

```diff
@@ -83,7 +85,7 @@ abstract class PartitioningAwareFileIndex(
           val files: Seq[FileStatus] = leafDirToChildrenFiles.get(path) match {
             case Some(existingDir) =>
               // Directory has children files in it, return them
-              existingDir.filter(f => matchGlobPattern(f) && isNonEmptyFile(f))
+              existingDir.filter(f => matchPathPattern(f) && isNonEmptyFile(f))
```

Comment:
`f` here is `FileStatus`. It looks inefficient to get the `FileStatus` again in https://github.com/apache/spark/pull/28841/files#diff-3c816e65d0bbdf16d46c7ea4e0b8cbaeR55

Reply:
> `f` here is `FileStatus`. It looks inefficient to get the `FileStatus` again in https://github.com/apache/spark/pull/28841/files#diff-3c816e65d0bbdf16d46c7ea4e0b8cbaeR55

I agree.  I considered this in one of my earlier commits actually.  There are two places where we handle options, here in _PartitioningAwareMemoryFileIndex_ and _InMemoryFileIndex_.  In both places, we derive from the _PathFilter_ class which exposes an overridable method called _accept_ which takes _Path_ as its only parameter.  

I could break with the abstraction and overload the method so that it would take a FileStatus directly.  I'll look into this unless there are any objections.

Reply:
I can put a wrapper around this to make the behavior more efficient.

Reply:
@cloud-fan This has been resolved.